specifications/cloud_assembly.md

```diff
@@ -0,0 +1,398 @@
+following sub-sections.
+
+##### The `aws` protocol
+Environments using the `aws` protocol have the following format:
+```
+aws://account/region
+╰┬╯   ╰──┬──╯ ╰──┬─╯
+ │       │       └─ The name of an AWS region (e.g: eu-west-1)
+ │       └───────── An AWS account ID (e.g: 123456789012)
+ └───────────────── AWS protocol specifier
+```
+
+### `Missing`
+[Drops](#drop) may require contextual information to be available in order to correctly participate in a
+*Cloud Assembly*. When information is missing, *Cloud Assembly* producers report the missing information by adding
+entries to the `missing` section of the [manifest document](#manifest-document). The values are [JSON] `object`s that
+**MUST** conform to the following schema:
+
+Key            |Type             |Required|Description
+---------------|-----------------|:------:|-----------
+`provider`     |`string`         |Required|A tag that identifies the entity that should provide the information.
+`props`        |`Map<string,any>`|Required|Properties that are required in order to obtain the missing information.
+
+### Digital Signature
+#### Signing
+*Cloud Assemblers* **SHOULD** support digital signature of *Cloud Assemblies*. When support for digital signature is
+present, *Cloud Assemblers*:
+* **MUST** require the user to specify which [PGP][RFC 4880] key should be used.
+
+##### Signing Algorithm
+<!--TODO-->
+
+#### Verifying
+Deployment systems **SHOULD** support verifying signed *Cloud Assemblies*. If support for signature verification is not
+present, a warning **MUST** be emitted when processing a *Cloud Assembly* that contains the `signature.asc` file.
+
+Deployment systems that support verifying signed *Cloud Assemblies*:
+* **SHOULD** allow the user to *require* that an assembly is signed. When this requirement is active, an error **MUST**
+  be returned when attempting to deploy an un-signed *Cloud Assembly*.
+* **MUST** verify the integrity and authenticity of signed *Cloud Assemblies* prior to attempting to load any file
+  included in it, except for `signature.asc`.
+  * An error **MUST** be raised if the *Cloud Assembly*'s integirty is not verified by the signature.
+  * An error **MUST** be raised if the [PGP][RFC 4880] key has expired according to the signature timestamp.
+  * An error **MUST** be raised if the [PGP][RFC 4880] key is known to have been revoked. Deployment systems **MAY**
+    trust locally available information pertaining to the key's validity.
+* **SHOULD** allow the user to specify a list of trusted [PGP][RFC 4880] keys.
+
+## Annex
+### Examples of Drops for the AWS Cloud
+#### `@aws-cdk/aws-cloudformation.StackDrop`
+A [*CloudFormation* stack][CFN Stack].
+
+##### Properties
+Property     |Type                |Required|Description
+-------------|--------------------|:------:|-----------
+`template`   |`string`            |Required|The assembly-relative path to the *CloudFormation* template document.
```

Comment:
`stackName` needs to be a parameter as well.

Or will you force `stackName` == logicalID of `drop`?

Reply:
Good point. I think I was implicitly assuming logicalId = stackName, but that feels brittle, especially if you want to computer-generate the logical ID's (the spec, after all, doesn't bind a particular meaning to them beyond "they identify a thing with the doc").